### PR TITLE
OSMorphing overhaul fixes for RHEL-based distros

### PR DESCRIPTION
**Pull request by Dany9966**

### CODE REVIEW

File: coriolis/osmorphing/redhat.py [hunk 1]
```diff
@@ -188,8 +187,13 @@ def _run_dracut(self):
         self._run_dracut_base('kernel')
 
     def _run_dracut_base(self, rpm_base_name):
-        package_names = self._exec_cmd_chroot(
-            'rpm -q %s' % rpm_base_name).decode().splitlines()
+        package_names = []
+        try:
+            package_names = self._exec_cmd_chroot(
+                'rpm -q %s' % rpm_base_name).decode().splitlines()
+        except exception.CoriolisException as ex:
```
Comment (aznashwan) [inline coriolis/osmorphing/redhat.py]: Is there any reason why we're only catching `CoriolisException`s here?
I'm pretty sure it's not even raised by `_exec_cmd_chroot` at all.
I'd catch anything inheriting from base `Exception`

File: coriolis/osmorphing/redhat.py [hunk 2]
```diff
@@ -188,8 +187,13 @@ def _run_dracut(self):
...
+            LOG.trace(ex)
```
Comment (aznashwan) [inline coriolis/osmorphing/redhat.py]: Considering the tools have a `self._event_manager`, I'd emit a clear warning like:
`Failed to query kernel package names. Unable to rebuild initrds for the new platform.`
NOTE: please double-check that the "kernel package names" part is indeed accurate with respect to every place `_run_dracut_base` is called.

File: coriolis/osmorphing/redhat.py [hunk 3]
```diff
@@ -57,11 +59,8 @@ def __init__(self, conn, os_root_dir, os_root_dev,
         self._enable_repos = []
 
     def disable_predictable_nic_names(self):
-        kernel_versions = self._list_dir("lib/modules")
-        for version in kernel_versions:
-            cmd = '/sbin/new-kernel-pkg --update --kernel-args="%s" %s'
-            self._exec_cmd_chroot(cmd % (
-                "net.ifnames=0 biosdevname=0", version))
+        cmd = 'grubby --update-kernel=ALL --args="%s"'
```
Comment (aznashwan) [inline coriolis/osmorphing/redhat.py:61]: Will this also work on the 6s?
Even if so, we should make sure to explicitly install it to be sure it's there.

**Review by aznashwan — COMMENTED** [review]
Needs checking some things.

File: coriolis/osmorphing/redhat.py (hunk 3 above)
Comment (Dany9966) [reply coriolis/osmorphing/redhat.py:61]: I checked a centos6 release, it was there. Will also add an install command so we make sure it's installed.

**Comment by aznashwan** [comment]
Merging, thanks!

**Pull request merged**